src/pages/instances/Events.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { FC, useEffect, useState } from "react";
+import { useInstanceLoading } from "context/instanceLoading";
+import { LxdEvent } from "types/event";
+
+const Events: FC = () => {
+  const instanceLoading = useInstanceLoading();
+  const [eventWs, setEventWs] = useState<WebSocket | null>(null);
+
+  const handleEvent = (event: LxdEvent) => {
+    if (event.metadata.action === "instance-created") {
+      const instanceName = event.metadata.source.split("/").pop() ?? "";
+      const entry = instanceLoading.get(instanceName);
+      if (entry && entry.type === "Creating" && entry.callback) {
+        entry.callback();
+      }
+    }
+  };
+
+  const connectEventWs = () => {
+    const wsUrl = `wss://${location.host}/1.0/events?type=operation,lifecycle`;
+    const ws = new WebSocket(wsUrl);
+    ws.onopen = () => {
+      setEventWs(ws);
+    };
+    ws.onclose = () => {
+      setEventWs(null);
+    };
+    ws.onmessage = (message: MessageEvent<Blob | string | null>) => {
+      if (typeof message.data !== "string") {
+        console.log("Invalid format on event api: ", message.data);
+        return;
+      }
+      const event = JSON.parse(message.data) as LxdEvent;
+      setTimeout(() => handleEvent(event), 1000); // handle with delay to allow operations to vanish
```

Comment:
Setting this timeout doesn't seem to be enough. Most of the times I tried I received the following error:

![Screenshot from 2023-05-29 21:15:37](https://github.com/canonical/lxd-ui/assets/56583786/706ec8da-d0d0-4c07-81dd-89953be2987f)

which makes me think that we get the "instance-created" event even earlier than 1 second before the "create" operation ceases to be "running". Ideally, we should check that the status of the "create" operation has transitioned from "running" to "success" - or even better, that the operation cannot be found anymore - before trying to start the instance.

Reply:
I saw this as well before, but with the latest implementation can't reproduce it any more even on our super slowish demo server. If it still happens, please reopen.